File: python/programmers/hash/전화번호목록.py
'''
* 🤷‍♂️ Created by wwlee94 on 2020.01.10
문제에서 이야기하는 '접두어'의 의미를 잘 파악해야함
문자열 중간에 포함되는 문자열은 해당되지 않고 문자열 시작 지점부터 일치하는 문자열만 해당된다.
'''
def solution(phone_book):
    leng = len(phone_book)
    phone_book = sorted(phone_book, key= len) # 길이에 따라 오름차순 정렬
    for i in range(leng-1):
        for j in range(i+1, leng):
            if phone_book[j].startswith(phone_book[i]): # 시작되는지 검증
                return False           
    return True

File: java/programmers/hash/전화번호목록.java
/*
* 🙆‍♂️ Created by wwlee94 on 2020.03.23
https://programmers.co.kr/learn/courses/30/lessons/42577
*/

import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        int length = phone_book.length;
        // 문자 길이로 오름차순
        Arrays.sort(phone_book, (s1, s2) -> s1.length() - s2.length());
        for(int i=0;i<length-1;i++){
            String base = phone_book[i];
            for(int j=i+1;j<length;j++){
                if(phone_book[j].startsWith(base)) answer = false;
            }
        }
        return answer;
    }
}